### PR TITLE
Add collection id to release, record, compiled_release: Stage 4

### PR DESCRIPTION
**Pull request by odscjames**
Convert existing data
Make columns NOT NULL

https://github.com/open-contracting/kingfisher-process/issues/236

### CODE REVIEW

**Comment by odscjames** [comment]
Technically we can do a straight deploy of this, it would take 2 days (tho it should be useable in that time).

To avoid that I will deploy by hand and comment out the updates first.

But including the updates  means that anyone else with installs of this app will be able to upgrade normally with no issues (except maybe, the time it takes).

File: ocdskingfisherprocess/maindatabase/migrations/versions/413c84a833f5_add_collection_id_2.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '413c84a833f5'
+down_revision = 'bdfc23d8334c'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute(
```
Comment (jpmckinney) [inline ocdskingfisherprocess/maindatabase/migrations/versions/413c84a833f5_add_collection_id_2.py]: @odscjames I assume we don't want to run these again, as they will take days?

**Comment by odscjames** [comment]
I think we both commented at the same time :-)

**Comment by jpmckinney** [comment]
> I think we both commented at the same time :-)

Indeed :) If you add the WHERE clause from my comment, I think this can run without the manual updates. If run after the INDEX is created, it'll be fast. If run before, it'll take a table scan, but I think that'd take less than an hour anyway (hopefully much less).

**Comment by odscjames** [comment]
If your happy with me doing a manual deploy, is it ok to leave them in? Adding WHERE would require extra testing, whilst we have tested what's here.

**Comment by odscjames** [comment]
We did it again; I was saying adding WHERE will take more testing but I can do that now if you want.

**Comment by jpmckinney** [comment]
Let's add the clauses (`AND record.collection_id IS NULL`, `AND release.collection_id IS NULL`, `AND compiled_release.collection_id IS NULL`). The logic is straightforward; I don't think a test is needed.

**Comment by jpmckinney** [comment]
Anyhow, if you prefer commenting out, that's fine too, so approving.

**Comment by odscjames** [comment]
I'll go with commenting out if that's ok.  That should have less unknowns when deploying - only slightly, but that still helps. Deploying now.

**Comment by odscjames** [comment]
Deploy cancelled. Another transaction/connection that I could not identify was blocking this from finishing. I was just leaving  it to run, but then I realised that the migration transaction was blocking other SELECT transactions on the database. At this point I stopped the migration - I have put a note on the token and will try again very soon.

**Comment by odscjames** [comment]
Never mind - the mystery connection just stopped. I ran the migration fine.

**Comment by jpmckinney** [comment]
FYI, If you login as root and:

```
sudo -u postgres psql postgres -c 'SELECT pid, state, wait_event_type, NOW() - query_start, query FROM pg_stat_activity;'
```

You can see which queries are active, and which are waiting for locks.

**Comment by odscjames** [comment]
Yeah, I traced it to an IP address in Germany but I could not identify what person / app was running it. No nice comments and I didn't recognise the SQL it was running as being from any of our code. I was just starting to ask people if anyone would claim responsibility for it when it went away by itself so  ¯\_(ツ)_/¯  Interesting thought tho; we could get some of our apps to put SQL comments in queries like pgadmin and redash do, then any poor data base admins can identify them in the future.

**Comment by odscjames** [comment]
[Review Kingfisher Process PRs](https://trello.com/c/UNsZ27w5/170-review-kingfisher-process-prs)

**Comment by jpmckinney** [comment]
I made a follow-up issue for Colaboratory notebooks and another for KF Views.

For this repo, for the https://github.com/open-contracting/kingfisher-process/commits/django branch, I found this open Django issue https://code.djangoproject.com/ticket/24638, which I don't expect to be closed anytime soon.

Otherwise, besides pgAdmin and Redash which you note already add comments, I don't know what other code is making queries.

**Pull request merged**